sns/parity_math.c

```diff
@@ -32,50 +32,195 @@
 #define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_SNS
 #include "lib/trace.h"
 
-#define ir_invalid_col_t UINT8_MAX
+enum {
+	IR_INVALID_COL = UINT8_MAX,
+};
+
+#define ALLOC_ARR_INFO(arr, nr, msg, ret) ({					\
+	M0_ALLOC_ARR(arr, nr);							\
+	if (arr == NULL) 							\
+		(ret) = M0_ERR_INFO(-ENOMEM,					\
+				    "failed to allocate memory for " msg);	\
+	else 									\
```

Comment:
Let's add trace log only for failure cases.

Reply:
Done.